src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Register/Query/RegisterQuerySearchAssessmentOrganisationsTests.cs

```diff
@@ -0,0 +1,124 @@
+        private AssessmentOrganisationSummary _assessmentOrganisationDetails2;
+
+        [SetUp]
+        public void Setup()
+        {
+            _registerQueryRepository = new Mock<IRegisterQueryRepository>();
+            _searchValidator = new Mock<IEpaOrganisationSearchValidator>();
+            _logger = new Mock<ILogger<SearchAssessmentOrganisationHandler>>();
+
+            _assessmentOrganisationDetails1 = new AssessmentOrganisationSummary { Id = "EPA9999", Name = "Name 100", Ukprn = 777777 };
+            _assessmentOrganisationDetails2 = new AssessmentOrganisationSummary { Id = "EPA8888", Name = "Name 10" };
+
+            _expectedOrganisationListOfDetails = new List<AssessmentOrganisationSummary>
+            {
+                _assessmentOrganisationDetails1,
+                _assessmentOrganisationDetails2
+            };
+          
+            _searchAssessmentOrganisationsHandler = new SearchAssessmentOrganisationHandler(_registerQueryRepository.Object, _searchValidator.Object, _logger.Object);
+        }
+
+        [Test]
+        public void SearchAssessmentOrganisationsThrowsBadRequestExceptionIfSearchStringTooShort()
+        {
+            var request = new SearchAssessmentOrganisationsRequest { Searchstring = "AB " };
+            Assert.ThrowsAsync<BadRequestException>(() => _searchAssessmentOrganisationsHandler.Handle(request, new CancellationToken())); 
+        }
+
+        [Test]
+        public void SearchAssessmentOrganisationsWithValidOrganisationId()
+        {
+            const string searchstring = "epacode";
+            var request = new SearchAssessmentOrganisationsRequest { Searchstring = searchstring };
+            _searchValidator.Setup(v => v.IsValidEpaOrganisationId(searchstring)).Returns(true);
+            _searchValidator.Setup(v => v.IsValidUkprn(searchstring)).Returns(true);
+            _registerQueryRepository.Setup(r => r.GetAssessmentOrganisationsByOrganisationId(searchstring))
+                .Returns(Task.FromResult(_expectedOrganisationListOfDetails.AsEnumerable()));
+            var organisations = _searchAssessmentOrganisationsHandler.Handle(request, new CancellationToken()).Result;
+
+            _searchValidator.Verify(v => v.IsValidEpaOrganisationId(searchstring));
+            _searchValidator.Verify(v => v.IsValidUkprn(It.IsAny<string>()),Times.Never);
+            _registerQueryRepository.Verify(r => r.GetAssessmentOrganisationsByOrganisationId(searchstring));
+            organisations.Count.Should().Be(2);
+            organisations.Should().Contain(_assessmentOrganisationDetails1);
+            organisations.Should().Contain(_assessmentOrganisationDetails2);
+        }
+
+        [Test]
+        public void SearchAssessmentOrganisationsWithValidukprn()
+        {
+            const string searchstring = "12345678";
+            var request = new SearchAssessmentOrganisationsRequest { Searchstring = searchstring };
```

Comment:
General comment but would seperate out asserts via AAA into seperate methods like many of the other tests, 
and maybe mimic GivenWhen syntax. Maybe we should discuss with everyone whether or not we need consistency with tests. I'm not sure for example whether or not we should have a template for tests or whether or not to consider Xunit for example?

Reply:
big discussion to be had another time :)

Reply:
👍